Sources/Swift/Integrations/Metrics/MetricBatcher.swift

```diff
@@ -0,0 +1,90 @@
+@_implementationOnly import _SentryPrivate
+import Foundation
+
+/// Protocol for batching metrics with scope-based attribute enrichment.
+protocol MetricBatcherProtocol {
```

Comment:
`l`: We already have the Metric**s**Integration. I would keep the naming consistend.
```suggestion
protocol MetricsBatcherProtocol {
```